pkg/services/ngalert/notifier/channels/webhook.go

```diff
@@ -21,98 +23,104 @@ import (
 // alert notifications as webhooks.
 type WebhookNotifier struct {
 	*Base
-	URL        string
-	HTTPMethod string
-	MaxAlerts  int
-	log        log.Logger
-	ns         notifications.WebhookSender
-	images     ImageStore
-	tmpl       *template.Template
-	orgID      int64
-
-	User     string
-	Password string
-
-	AuthorizationScheme      string
-	AuthorizationCredentials string
+	log       log.Logger
+	ns        notifications.WebhookSender
+	images    ImageStore
+	tmpl      *template.Template
+	orgID     int64
+	maxAlerts int
+	settings  webhookSettings
 }
 
-type WebhookConfig struct {
-	*NotificationChannelConfig
-	URL        string
-	HTTPMethod string
-	MaxAlerts  int
+type webhookSettings struct {
+	URL        string      `json:"url,omitempty" yaml:"url,omitempty"`
+	HTTPMethod string      `json:"httpmethod,omitempty" yaml:"httpmethod,omitempty"`
+	MaxAlerts  interface{} `json:"maxAlerts,omitempty" yaml:"maxAlerts,omitempty"`
 
 	// Authorization Header.
-	AuthorizationScheme      string
-	AuthorizationCredentials string
+	AuthorizationScheme      string `json:"authorization_scheme,omitempty" yaml:"authorization_scheme,omitempty"`
+	AuthorizationCredentials string `json:"authorization_credentials,omitempty" yaml:"authorization_credentials,omitempty"`
 	// HTTP Basic Authentication.
-	User     string
-	Password string
+	User     string `json:"username,omitempty" yaml:"username,omitempty"`
+	Password string `json:"password,omitempty" yaml:"password,omitempty"`
+}
+
+func buildWebhookSettings(factoryConfig FactoryConfig) (webhookSettings, error) {
+	settings := webhookSettings{}
+	err := factoryConfig.Config.unmarshalSettings(&settings)
+	if err != nil {
+		return settings, fmt.Errorf("failed to unmarshal settings: %w", err)
+	}
+	if settings.URL == "" {
+		return settings, errors.New("required field 'url' is not specified")
+	}
+	if settings.HTTPMethod == "" {
+		settings.HTTPMethod = http.MethodPost
+	}
+	settings.User = factoryConfig.DecryptFunc(context.Background(), factoryConfig.Config.SecureSettings, "username", settings.User)
```

Comment:
Previously user was not encrypted, only password. I don't think we need to decrypt